framework/decode/dx12_replay_consumer_base.cpp

```diff
@@ -4532,32 +4549,34 @@ void Dx12ReplayConsumerBase::PostCall_ID3D12Device_CreateShaderResourceView(
     auto heap_object_info = GetObjectInfo(DestDescriptor.heap_id);
     auto heap_extra_info  = GetExtraInfo<D3D12DescriptorHeapInfo>(heap_object_info);
 
-    ShaderResourceInfo info;
-    info.resource_id   = pResource;
-    info.replay_handle = *DestDescriptor.decoded_value;
+    DHCbvSrvUavInfo info;
+    info.type              = D3D12_DESCRIPTOR_RANGE_TYPE_SRV;
+    auto& srv_info         = info.srv;
+    srv_info.resource_id   = pResource;
+    srv_info.replay_handle = *DestDescriptor.decoded_value;
     if (pDesc->IsNull())
     {
-        info.is_view_null = true;
-        info.subresource_indices.emplace_back(0);
+        srv_info.is_desc_null = true;
+        srv_info.subresource_indices.emplace_back(0);
     }
     else
     {
-        info.view         = *(pDesc->GetMetaStructPointer()->decoded_value);
-        info.is_view_null = false;
+        srv_info.desc     = *(pDesc->GetMetaStructPointer()->decoded_value);
+        srv_info.is_desc_null = false;
 
         if (pResource != format::kNullHandleId)
         {
             auto desc        = reinterpret_cast<ID3D12Resource*>(GetObjectInfo(pResource)->object)->GetDesc();
             auto mip_count   = desc.MipLevels;
             auto array_count = desc.DepthOrArraySize;
-            switch (info.view.ViewDimension)
+            switch (srv_info.desc.ViewDimension)
```

Comment:
Is `subresource_indices` for all the `decode::DH*Info` classes used only for dump resources? If so, what do you think of moving all the code to find `subresource_indices` to the `Dx12DumpResources` class in order to keep non-dump resources code paths from doing additional work?

For example, could this switch statement be moved to a new static function such as `Dx12DumpResources::GetDescriptorSubresourceIndices(DHShaderResourceViewInfo& srv_info);` and call it here: `Dx12DumpResources::GetDescriptorSubresourceIndices(srv_info.desc);`? And same thing for the other descriptor types that have `subresource_indicies`